rumqttc/examples/async_manual_acks_v5.rs

```diff
@@ -10,6 +10,7 @@ fn create_conn() -> (AsyncClient, EventLoop) {
     let mut mqttoptions = MqttOptions::new("test-1", "localhost", 1884);
     mqttoptions
         .set_keep_alive(Duration::from_secs(5))
+        .set_session_expiry_interval(u32::MAX.into())
```

Comment:
we only want to show manual acks in these example, why we need session expiry interval to be u32 max?

Reply:
According to section 3.1.11.2 in https://docs.oasis-open.org/mqtt/mqtt/v5.0/mqtt-v5.0.pdf
```
720 If the Session Expiry Interval is 0xFFFFFFFF (UINT_MAX), the Session does not expire.
721
722 The Client and Server MUST store the Session State after the Network Connection is closed if the 
723 Session Expiry Interval is greater than 0 [MQTT-3.1.2-23].
```

Server stores state only if session expiry interval is greater than 0, so on connect it must be provided to broker. We want the connection session does not expire, so we set the expiry interval to UINT_MAX (u32 max) here.

Reply:
We probably should not make the change within this particular example as `u32::MAX` should be the default value?

Reply:
The default is absent, the value 0 is used, in this case broker does not store the session state.
```
717 If the Session Expiry Interval is absent the value 0 is used. If it is set to 0, or is absent, the Session ends 
718 when the Network Connection is closed.
```

Reply:
Understood, this makes sense, LGTM @swanandx